libethashseal/Ethash.cpp

```diff
@@ -166,12 +166,6 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 		else
 			_t.checkChainId(-4);
 	}
-	if (_ir & ImportRequirements::TransactionBasic && _t.baseGasRequired(evmSchedule(_header.number())) > _t.gas())
```

Comment:
Do you mean you would like to have these checks to be executed for `NoProof`, too? Then would it be possible to move it into base class, to `SealEngineFace::verifyTransaction()` ?

The problem with moving it into `Executive` is that these checks won't be done until transaction is being executed, and we would like to reject pending transactions that don't satisfy this earlier, before adding them into pending block.

Reply:
I think this is a consensus rule that it is not allowed to "accept" transactions that could exceed block gas limit. I'd prefer to move it somewhere else than Executive. Executive is a messy part that does not do anything useful per se.

Reply:
we decided with @pirapira that it does not related to sealing. @pirapira ?

Reply:
Actually I was not right, we don't call this for pending transactions, because these checks depend on the block header params, so don't make sense until we include tx into block

Reply:
But all other checks in `verifyTransaction` are not related to sealing either. I think it was put there because all these checks require `ChainOperationParams`

Reply:
`SealEngineFace` is fine for me.  That's the place for things not dependent on any particular sealing.

Reply:
But still: before this change we were able to reject incoming blocks containing invalid transactions earlier (in `BlockQueue` validation thread) than we begin to try to execute and import them

Reply:
without performing those checks for NoProof engine a transaction with negative gas is executed.

Reply:
some transaction tests are now failing it seems to be. but is it because cpp implementation was wrong without those checks?

Reply:
I think transaction tests validate transactions using `SealEngine` and now you removed some checks, so it's failing